lib/pages/account.dart

```diff
@@ -0,0 +1,442 @@
+  GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  AccountRepository _accountRepo = AccountRepository.getInstance();
+  TextStyle _style;
+  Future<Account> accountInfo;
+  Account _account;
+  AuthArguments _args;
+
+  TextEditingController _firstNameController;
+  TextEditingController _lastNameController;
+  TextEditingController _phoneController;
+  GlobalKey<FormState> _formKey;
+
+  @override
+  initState() {
+    super.initState();
+//    _accountRepo.invalidateCurrentAccount();
+
+    _style = getAppTheme().primaryTextTheme.body1;
+    accountInfo = _accountRepo.getAccount();
+    _formKey = GlobalKey<FormState>();
+
+    accountInfo.then((Account account) async {
+      setState(() {
+        _account = account;
+      });
+
+  }).catchError((error) {
+      debugPrint('Error getting user info');
+
+      SnackBar snackBar = SnackBar(
+      content: Text('Could not connect to server'),
+    );
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+    _scaffoldKey.currentState.showSnackBar(snackBar);
+    });
+
+  });
+
+
+
+  }
+
+  void submit() {
+
+
+    String email = _account.email;
+    String firstName = _firstNameController.text;
+    String lastName = _lastNameController.text;
+    String phone = _phoneController.text;
+    print("Submit function called");
+
+    _account.firstName = firstName;
+    _account.lastName = lastName;
+    _account.phoneNumber = phone;
+
+
+    if (_formKey.currentState.validate()) {
+      appAuth.updateAccount(email, firstName, lastName, phone).then((result) {
+        if (result != true) {
+          print("Error");
+          Widget okButton = FlatButton(
+            child: Text("OK"),
+            onPressed: () {
+              Navigator.of(context).pop();
+            },
+          );
+
+          AlertDialog alert = AlertDialog(
+            title: Text("Error"),
+            content: Text("Your settings have not been saved"),
+            actions: [
+              okButton,
+            ],
+          );
+
+          showDialog(context: context,
+            builder: (BuildContext context){
+              return alert;
+            },
+          );
+
+        } else {
+          Widget okButton = FlatButton(
+            child: Text("OK"),
+            onPressed: () {
+              Navigator.of(context).pop();
+            },
+          );
+
+          AlertDialog alert = AlertDialog(
+            title: Text("Settings"),
+            content: Text("Your settings have been saved successfully"),
+            actions: [
+              okButton,
+            ],
+          );
+
+          showDialog(context: context,
+          builder: (BuildContext context){
+            return alert;
+          },
+          );
+
+         // Navigator.of(context).pushReplacementNamed('/home');
+        }
+      });
+    }
+  }
+
+
+  @override
+  Widget build(BuildContext context) {
+
+    _firstNameController = TextEditingController(text: _account.firstName);
```

Comment:
This usually gets called before the API call to get account info has returned, meaning account is null here. In debug mode, this causes a few frames of an error message, in release mode, the app may crash. this can be resolved using a FutureBuilder, the home page and the history page both have examples on how to use this